addons/account/report/account_followup_report.py

```diff
@@ -0,0 +1,261 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Odoo, Open Source Management Solution
+#    Copyright (C) 2004-2014 OpenErp S.A. (<http://odoo.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, tools
+from datetime import datetime
+from hashlib import md5
+from openerp.tools.misc import formatLang
+
+
+class report_account_followup_report(models.AbstractModel):
+    _name = "account.followup.report"
+    _description = "Followup Report"
+
+    @api.model
+    def get_lines(self, context_id, line_id=None, public=False):
+        lines = []
+        domain = [('partner_id', '=', context_id.partner_id.id), ('reconciled', '=', False), ('account_id.deprecated', '=', False), ('account_id.internal_type', '=', 'receivable')]
+        if public:
+            domain.append(('blocked', '=', False))
+        total = 0
+        total_issued = 0
+        amls = self.env['account.move.line'].search(domain)
+        currencies = self.env['res.currency'].browse([])
+        for aml in amls:
+            currencies |= aml.currency_id or aml.company_id.currency_id
+        for currency in currencies:
```

Comment:
Why 2 for loops?  1 would be nicer... with a dictionary key currency and values the lines